### PR TITLE
feat(metrics): add bpm controlled processes metrics

### PR DESCRIPTION
**Pull request by TangliziGit**
### What problem does this PR solve?

> Problem Summary: a part of #2397

### What's changed and how it works?

> What's Changed:
1. add `bpm.GetControlledProcessIDs` method to query the process IDs controlled by background process manager, and add a testcase for `bpm.GetControlledProcessIDs`.
2. add `bpmControlledProcessTotal` and `bpmControlledProcesses` metrics.

### Related changes

- [ ] Need to update `chaos-mesh/website`
- [ ] Need to update `Chaos Dashboard`
- [ ] Need to **cheery-pick to release branches**

### Checklist

Tests

- [x] Unit test
- [ ] E2E test
- [x] Manual test (add detailed scripts or steps below)
 - create HTTPChaos to trigger `bpm.StartProcess`, then observe the `/metrics` endpoint to see if there are corresponding metrics.
- [ ] No code

Side effects

- [ ] Breaking backward compatibility

### Release note 

```text
Please add a release note.

You can safely ignore this section if you don't think this PR needs a release note.
```

### CODE REVIEW

**Review by yeya24 — COMMENTED** [review]
I am not sure why do we need bpm process metrics? Can you please give an example? I don't see the benefits or usecase of having it.

**Comment by TangliziGit** [comment]
> I am not sure why do we need bpm process metrics? Can you please give an example? I don't see the benefits or usecase of having it.

In order to visually observe the health status of `chaos-daemon`. For example, when a Schedule containing IOChaos is submitted and the IOChaos executes, if the target folder does not exist, the execution of IOChaos will fail due to mounting failure. At the same time, the reconciler will always retry to execute the same error. If there are other types of chaos that need to be injected into the same container using BPM at the same time, it may cause a mess because the BPM identifiers lock is not released.

Of course, you can use the log to notice the execution failure, but more intuitive metrics may be more helpful: `chaos_daemon_bpm_controlled_process_total` will grow rapidly over a period of time, while `chaos_daemon_bpm_controlled_processes` will not change.

File: pkg/bpm/bpm.go [hunk 1]
```diff
@@ -301,6 +315,26 @@ func (m *BackgroundProcessManager) Stdio(pid int, startTime int64) *Stdio {
...
+	pid := os.Getpid()
+	for _, p := range processes {
+		if pid == p.PPid() {
+			res = append(res, p.Pid())
+		}
+	}
+
+	log.Info("all controlled processes has been scanned", "PID", res)
+	return res, nil
+}
```
Comment (STRRL) [inline pkg/bpm/bpm.go:336]: This implementation may introduce more processes that might be not controller by this bpm instance. You are finding that all the processes that controller by **this process**, but one process might use more than one instance of `bpm` and has other logic that spawns new processes. 

I suggest that use `identifiers *sync.Map` in struct `BackgroundProcessManager`, the key of this map store the pid.

File: pkg/metrics/chaos_daemon.go [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+type ChaosDaemonMetricsCollector struct {
+	backgroundProcessManager  bpm.BackgroundProcessManager
+	bpmControlledProcessTotal prometheus.Counter
+	bpmControlledProcesses    prometheus.Gauge
+}
+
+// NewChaosDaemonMetricsCollector initializes metrics for each chaos daemon
+func NewChaosDaemonMetricsCollector() *ChaosDaemonMetricsCollector {
+	return &ChaosDaemonMetricsCollector{
+		backgroundProcessManager: bpm.NewBackgroundProcessManager(nil),
```
Comment (STRRL) [inline pkg/metrics/chaos_daemon.go:36]: As the former comments say, you need to use another way to assembly this `ChaosDaemonMetricsCollector` with the certain in-use `bpm` instance. :D

**Review by STRRL — CHANGES_REQUESTED** [review]
PTAL @TangliziGit

File: pkg/bpm/bpm.go (hunk 1 above)
Comment (TangliziGit) [reply pkg/bpm/bpm.go:336]: OK, I will modify it later.

File: pkg/metrics/chaos_daemon.go (hunk 2 above)
Comment (TangliziGit) [reply pkg/metrics/chaos_daemon.go:36]: OK, I got it.

File: pkg/bpm/bpm.go (hunk 1 above)
Comment (TangliziGit) [reply pkg/bpm/bpm.go]: Done.

**Review by STRRL — APPROVED** [review]

**Comment by STRRL** [comment]
/cc @YangKeao @Hexilee @iguoyr @Andrewmatilde 

PTAL

**Review by cwen0 — COMMENTED** [review]
Rest LGTM

**Review by YangKeao — APPROVED** [review]
LGTM

**Pull request merged**